src/ModelContextProtocol/Server/AIFunctionMcpServerTool.cs

```diff
@@ -60,6 +61,14 @@ internal sealed class AIFunctionMcpServerTool : McpServerTool
             options);
     }
 
+    // TODO: Fix the need for this suppression.
+    [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2111:ReflectionToDynamicallyAccessedMembers",
+        Justification = "AIFunctionFactory ensures that the Type passed to AIFunctionFactoryOptions.CreateInstance has public constructors preserved")]
```

Comment:
@eiriktsarpalis, this is what I pinged you about offline. I think this is safe? But regardless it's not ideal. How do we fix this? Custom delegate type in MEAI? Something else?